Assets/Scripts/Character/ViewBob.cs

```diff
@@ -0,0 +1,137 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using MyBox;
+
+// TODO: Add randomness
+// TODO: Add rotational inertia
+// TODO: add jump drift
+
+public class ViewBob : MonoBehaviour {
+    [SerializeField] [MustBeAssigned] private CharacterMovement movement;
+    [SerializeField] [MustBeAssigned] private Transform armsTransform;
+    [SerializeField] [MustBeAssigned] private Transform cameraTransform;
+
+    [SerializeField] [PositiveValueOnly] private float breatheIntensity = 1;
+    [SerializeField] [PositiveValueOnly] private float breatheSpeed = 1;
+    [SerializeField] [PositiveValueOnly] private float bobIntensity = 1;
+    [SerializeField] [PositiveValueOnly] private float bobSpeed = 1;
+    private float dragTransition = 0.05f; // how fast the arm pulls back when you begin moving
+    private float dragMax = -0.06f; // how much the arm pulls back when you begin moving
+
+    [SerializeField] [ReadOnly] private float dragOffset = 0;
+    [SerializeField] [ReadOnly] private float bobVerticalOffset = 0;
+    [SerializeField] [ReadOnly] private float bobLeftOffset = 0;
+    [SerializeField] [ReadOnly] private float bobRightOffset = 0;
+    [SerializeField] [ReadOnly] private float breatheOffset = 0;
+    private Coroutine currentBobVerticalRoutine;
+    private Coroutine currentBobLeftRoutine;
+    private Coroutine currentBobRightRoutine;
+    private Coroutine currentBreatheRoutine;
+
+    private Vector3 cameraInitialLocalPosition;
+    private Vector3 armsInitialLocalPosition;
+    public bool isEnabled = true;
+    public float GLOBAL_BOB_SPEED_MULTIPLIER; // bob speed multiplier for sprinting
+
+    void Start() {
+        cameraInitialLocalPosition = cameraTransform.localPosition;
+        armsInitialLocalPosition = armsTransform.localPosition;
+    }
+
+    IEnumerator BobVertical() {
+        float INTENSITY = bobIntensity * 0.02f;
+        float SPEED = bobSpeed * 7f * GLOBAL_BOB_SPEED_MULTIPLIER;
+
+        float x = 0;
+        while (x <= Mathf.PI) {
+            bobVerticalOffset = -Mathf.Abs(Mathf.Sin(x)) * INTENSITY;
+
+            yield return null; //Waits/skips one frame
+
+            x += Time.deltaTime * SPEED;
+        }
+
+        bobVerticalOffset = 0;
+        currentBobVerticalRoutine = null;
+    }
+
+    IEnumerator BobLeft() {
+        float INTENSITY = bobIntensity * 0.02f;
+        float SPEED = bobSpeed * 7f * GLOBAL_BOB_SPEED_MULTIPLIER;
+
+        float x = 0;
+        while (x <= Mathf.PI) {
+            bobLeftOffset = -Mathf.Abs(Mathf.Sin(x)) * INTENSITY;
+
+            yield return null; //Waits/skips one frame
+
+            x += Time.deltaTime * SPEED;
+        }
+
+        bobLeftOffset = 0;
+        currentBobLeftRoutine = null;
+    }
+
+    IEnumerator BobRight() {
+        float INTENSITY = bobIntensity * 0.02f * -1; // negative sign here
```

Comment:
third and last time I'll mention it, haha. But instead of 0.02f I think we could have a multiplier variable for example